velox/common/base/BitUtil.cpp

```diff
@@ -118,13 +118,16 @@ void scatterBits(
     return;
   }
 #ifdef __BMI2__
-  int32_t highByte = numTarget / 8;
-  int32_t highBit = numTarget & 7;
-  int lowByte = std::max(0, highByte - 7);
+  uint32_t highByte = numTarget / 8;
+  uint32_t highBit = numTarget & 7;
+  uint32_t lowByte = 0;
+  if (highByte >= 7) {
```

Comment:
Can we check we generate optimal instruction for this?  Should not introduce any branching.  Maybe just use `?:` to be sure